backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/JsonRetrieverService.java

```diff
@@ -988,4 +1054,26 @@ private JsonProductBPartner toJsonProductBPartner(final I_C_BPartner_Product rec
 				//
 				.build();
 	}
+
+	private static JsonResponseBPBankAccount toJson(@NonNull final BPartnerBankAccount bankAccount)
+	{
+		final JsonMetasfreshId metasfreshId = JsonMetasfreshId.of(BPartnerBankAccountId.toRepoId(bankAccount.getId()));
+		final JsonMetasfreshId metasfreshBPartnerId = JsonMetasfreshId.of(BPartnerId.toRepoId(bankAccount.getId().getBpartnerId()));
+		final JsonMetasfreshId currencyId = JsonMetasfreshId.of(CurrencyId.toRepoId(bankAccount.getCurrencyId()));
```

Comment:
consider introducing JsonMetasfreshId.ofNullble(RepoIdAware)

Reply:
yeah, I think we actually need a different approach here, bc those ids are in fact always there, will update the code in that direction